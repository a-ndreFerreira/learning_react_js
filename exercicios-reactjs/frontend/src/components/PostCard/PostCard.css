.post {
    background-color: #fff;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);
    transition: transform .2s ease;
    border-radius: .5rem;
}

.post:hover {
    transform: scale(1.05);
}

.post img {
    max-width: 100%;
    border-radius: .5rem .5rem 0 0;
}

.post-content {
    padding: 1rem;
}