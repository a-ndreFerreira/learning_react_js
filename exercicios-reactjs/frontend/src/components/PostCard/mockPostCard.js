export const mockPostCard = {
    title: 'title 13',
    body: 'body',
    id: 1,
    cover: 'img/img.png'
}

//DO NOT EDIT THIS FILE