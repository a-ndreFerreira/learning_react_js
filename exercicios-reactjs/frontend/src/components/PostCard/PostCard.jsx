import './PostCard.css'

const PostCard = ({ cover, title, body, id }) => {
    return (
        <div className='post'>
            <img src={cover} alt={title} />
            <div className='post-content'>
                <h2>
                    {title} #{id}
                </h2>
                <p>
                    {body}
                </p>
            </div>
        </div>
    )
}

export default PostCard