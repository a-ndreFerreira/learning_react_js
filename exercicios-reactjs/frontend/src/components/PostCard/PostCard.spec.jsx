import { describe, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import PostCard from './PostCard';

import { mockPostCard } from './mockPostCard'

//mock de dados
const mock = mockPostCard;

describe('<PostCard />', () => {

    it('should image, heading h2 with id and body in screen', () => {

        // const {debug} = render(<PostCard {...mock} />); como se fosse um console.log do componente

        render(<PostCard {...mock} />);

        //debug(); como se fosse um console.log do componente

        //para ver se tem o documneto
        // expect(expectVar).toBeInTheDocument();

        expect(screen.getByRole('img', { name: 'title 13' }))
            .toHaveAttribute('src', 'img/img.png');

        expect(screen.getByRole('heading', { name: 'title 13 #1' }))
            .toBeInTheDocument();

        expect(screen.getByText('body')).toBeInTheDocument();

    })

    it('should match snapshot', () => {
        const { container } = render(<PostCard {...mock} />);

        expect(container.firstChild).toMatchSnapshot();

    })
})