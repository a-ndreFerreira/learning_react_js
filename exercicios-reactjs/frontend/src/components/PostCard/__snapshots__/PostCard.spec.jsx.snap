// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PostCard /> > should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title 13"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      title 13
       #
      1
    </h2>
    <p>
      body
    </p>
  </div>
</div>
`;
